#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_mips.dpatch by Wolfram Quester <wolfi@sigxcpu.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch to make inkscape build again with newer versions of libpng12.
## DP: 
## DP: We simply move the include <png.h> up until the cimpilation works in
## DP: all failing files.

@DPATCH@

diff -ur orig/inkscape-0.46/src/dialogs/export.cpp inkscape-0.46/src/dialogs/export.cpp
--- orig/inkscape-0.46/src/dialogs/export.cpp	2008-03-11 05:19:54.000000000 +0100
+++ inkscape-0.46/src/dialogs/export.cpp	2009-04-20 17:34:33.000000000 +0200
@@ -20,6 +20,8 @@
 # include "config.h"
 #endif
 
+#include <png.h>
+
 #include <gtk/gtk.h>
 #include <gtkmm/box.h>
 #include <gtkmm/buttonbox.h>
@@ -60,7 +63,6 @@
 #include "io/sys.h"
 
 #include "helper/png-write.h"
-#include <png.h>
 
 
 #define SP_EXPORT_MIN_SIZE 1.0
diff -ur orig/inkscape-0.46/src/main.cpp inkscape-0.46/src/main.cpp
--- orig/inkscape-0.46/src/main.cpp	2008-03-11 05:20:50.000000000 +0100
+++ inkscape-0.46/src/main.cpp	2009-04-20 18:18:58.000000000 +0200
@@ -51,6 +51,7 @@
 
 #include "gc-core.h"
 
+#include <png.h>
 #include "macros.h"
 #include "file.h"
 #include "document.h"
@@ -111,7 +112,6 @@
 
 #include "main-cmdlineact.h"
 
-#include <png.h>
 #include <errno.h>
 
 enum {
diff -ur orig/inkscape-0.46/src/sp-image.cpp inkscape-0.46/src/sp-image.cpp
--- orig/inkscape-0.46/src/sp-image.cpp	2008-03-11 05:19:56.000000000 +0100
+++ inkscape-0.46/src/sp-image.cpp	2009-04-20 17:24:12.000000000 +0200
@@ -29,6 +29,7 @@
 #include "enums.h"
 #include "attributes.h"
 
+#include <png.h>
 #include "print.h"
 #include "brokenimage.xpm"
 #include "document.h"
@@ -41,7 +42,6 @@
 #include "libnr/nr-matrix-fns.h"
 
 #include "io/sys.h"
-#include <png.h>
 #if ENABLE_LCMS
 #include "color-profile-fns.h"
 #include "color-profile.h"
